### ISSUE TITLE
Improve a11y within SchemaRow.svelte

### ISSUE THREAD

**Issue by seancolsen** [centerofci/mathesar#1840]
## Current behavior

- Within SchemaRow.svelte, we have an `<a>` element that contains a `DropdownMenu` that (eventually) contains a `<button>`.
- Nesting a `<button>` inside an `<a>` is invalid HTML and can lead to accessibility problems.

## Desired behavior

- Instead, we should absolutely position an `<a>` overlay across the entire SchemaRow area to serve as a click surface. It should have `aria-label` set to the schema name and should be positioned with a lower z-index than the DropdownMenu. This will also eliminate the need for `e.preventDefault();` within `Dropdown.svelte`.

**Comment by seancolsen** [comment]
This references code which is not yet merged as part of #1793

**Comment by himanshuc3** [comment]
I see there's a linked pr. Does that solve this issue or is it just an indirect improvement on which this issue builds upon?

**Comment by DanyaalMahmood** [comment]
Hey @seancolsen,  I'd like to work to work on this issue if it is still open.

**Comment by kgodey** [comment]
This is still open, please feel free to submit a PR.

**Comment by hemant-hc** [comment]
Hi @kgodey @seancolsen I have created a fix for this can you please check https://github.com/centerofci/mathesar/pull/2591

**Comment by Anish9901** [comment]
@hemant-hc Assigning this issue to you as you already have a PR open, but please be sure to get an issue assigned to you before working on it to avoid any conflicts.

**Comment by hemant-hc** [comment]
@Anish9901 Sure, will do that the next time.
